File: 1.Complete excercises code/kotlin/ example2.kt
fun arrayReplace(inputArray: MutableList<Int>, elemToReplace: Int, substitutionElem: Int): MutableList<Int> {
    for (i in 0 until inputArray.size) {
        if (inputArray[i] == elemToReplace) {
            inputArray[i] = substitutionElem
        }
    }
    return inputArray
}

fun main() {
    val inputArray = mutableListOf(1, 2, 1)
    val elemToReplace = 1
    val substitutionElem = 3
    val result = arrayReplace(inputArray, elemToReplace, substitutionElem)
    println(result) // Output: [3, 2, 3]
}

File: 1.Complete excercises code/dart/ example2.dart
List<int> arrayReplace(List<int> inputArray, int elemToReplace, int substitutionElem) {
  for (int i = 0; i < inputArray.length; i++) {
    if (inputArray[i] == elemToReplace) {
      inputArray[i] = substitutionElem;
    }
  }
  return inputArray;
}

void main() {
  List<int> inputArray = [1, 2, 1];
  int elemToReplace = 1;
  int substitutionElem = 3;
  List<int> result = arrayReplace(inputArray, elemToReplace, substitutionElem);
  print(result); // Output: [3, 2, 3]
}
